README.md

```diff
@@ -9,37 +11,111 @@ capabilities:
 It has been developed using **Golang** and
 **[Kubebuilder](https://book.kubebuilder.io/quick-start.html)**.
 
-## Implementation detail
+## Getting Started
 
+### Prerequisites
+- go version v1.22.0+
+- docker version 17.03+.
+- kubectl version v1.11.3+.
+- Access to a Kubernetes v1.11.3+ cluster.
 
+### To Deploy on the cluster
+**Build and push your image to the location specified by `IMG`:**
 
-## Developer docs
+```sh
+make docker-build docker-push IMG=<some-registry>/odh-model-controller:tag
+```
+
+**NOTE:** This image ought to be published in the personal registry you specified.
+And it is required to have access to pull the image from the working environment.
+Make sure you have the proper permission to the registry if the above commands don’t work.
+
+**Install the CRDs into the cluster:**
+
+```sh
+make install
+```
 
-Follow the instructions below if you want to extend the controller
-functionality:
+**Deploy the Manager to the cluster with the image specified by `IMG`:**
 
-### Run unit tests
+```sh
+make deploy IMG=<some-registry>/odh-model-controller:tag
+```
 
-Unit tests have been developed using the [**Kubernetes envtest
-framework**](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest).
+> **NOTE**: If you encounter RBAC errors, you may need to grant yourself cluster-admin
+privileges or be logged in as admin.
 
-Run the following command to execute them:
+**Create instances of your solution**
+You can apply the samples (examples) from the config/sample:
 
-```shell
-make test
+```sh
+kubectl apply -k config/samples/
 ```
 
-### Deploy local changes
+>**NOTE**: Ensure that the samples has default values to test it out.
 
-Build a new image with your local changes and push it to `<YOUR_IMAGE>` (by
-default `quay.io/${USER}/odh-model-controller:latest`).
+### To Uninstall
+**Delete the instances (CRs) from the cluster:**
 
-```shell
-make -e IMG=<YOUR_IMAGE> container-build container-push
+```sh
+kubectl delete -k config/samples/
 ```
 
-Deploy the manager using the image in your registry:
+**Delete the APIs(CRDs) from the cluster:**
+
+```sh
+make uninstall
+```
+
+**UnDeploy the controller from the cluster:**
+
+```sh
+make undeploy
+```
+
+## Project Distribution
+
+Following are the steps to build the installer and distribute this project to users.
+
+1. Build the installer for the image built and published in the registry:
+
+```sh
+make build-installer IMG=<some-registry>/odh-model-controller:tag
+```
+
+NOTE: The makefile target mentioned above generates an 'install.yaml'
+file in the dist directory. This file contains all the resources built
+with Kustomize, which are necessary to install this project without
+its dependencies.
+
+2. Using the installer
+
+Users can just run kubectl apply -f <URL for YAML BUNDLE> to install the project, i.e.:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/<org>/odh-model-controller/<tag or branch>/dist/install.yaml
+```
+
+## Contributing
```

Comment:
Currently, we don't have a CONTRIBUTING file. But I do agree with you.

However, as I mentioned in [my other comment](https://github.com/opendatahub-io/odh-model-controller/pull/324#discussion_r1887407176), I didn't want to delay the migration because of getting our READMES right.

Reply:
make sense, let's do it in a followup pr.

Reply:
Logged here: https://issues.redhat.com/browse/RHOAIENG-17149